### ISSUE TITLE
twisted.test.test_import is redundant

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2053]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2053|
|Type|defect|
|Created|2006-09-02 20:12:56Z|
|Branch|https://github.com/twisted/twisted/tree/remove-test-import-2053|

We used to have pretty crudtastic test coverage.  Having some tests that at least imported some modules that might not have been imported anyplace else was useful.

Is this still the case?  If so, we seriously need to write some tests.  If not, we should get rid of test_import, because it serves no purpose.

Attachments:

* [2503.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/5ca/5ca4045f83677d68114a72586591727fefe7b0c2/f547e8c2b8a380d90617786abe4acee5e89f05ef.patch) (2561 bytes) - added by ewong on 2013-11-05 05:38:27Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__2053 2053
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_remove_test_import_2053 branches/remove-test-import-2053
branch_author__lvh lvh
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1157227976000000 1157227976000000
changetime__1383754909000000 1383754909000000
version__None None
owner__lvh lvh
cc__therve cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_1"></a>|@therve commented|
|-|-|

I had a quick look at this, and unfortunately there is at least one module inside test_import without unittests: twisted.python.otp. Maybe we can still remove these tests and create a ticket for adding tests to otp?

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

What's the decision on this ticket?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

`twisted.python.otp` is deprecated (since 9.0, it seems, though it misreports as 8.3).  So we can get rid of that module and then presumably we can get rid of `test_import`, supposing that we have coverage for everything else it is importing.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra commented|
|-|-|

Replying to [exarkun](#note_4):
> `twisted.python.otp` is deprecated (since 9.0, it seems, though it misreports as 8.3).  So we can get rid of that module

I've opened [#5493](https://github.com/twisted/twisted/issues/5493) for this.

**Comment by twisted-trac** [comment]
|<img alt="ewong's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|ewong commented|
|-|-|

I've gone through test_import.py, and found that twisted.spread.flavors doesn't have coverage.
(Assuming that I've done it right.)  Do I remove the other tests and submit a patch, or do I just leave this as-is?

**Comment by twisted-trac** [comment]
|<img alt="ewong's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_7"></a>|ewong commented|
|-|-|

Replying to [ewong](#note_6):
> I've gone through test_import.py, and found that twisted.spread.flavors doesn't have coverage.
> (Assuming that I've done it right.)  Do I remove the other tests and submit a patch, or do I just leave this as-is?

Sorry.  twisted.spread.flavors is covered.

**Comment by twisted-trac** [comment]
|<img alt="ewong's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|ewong set owner to ewong<br>ewong set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="ewong's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|ewong removed owner<br>ewong set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)|@lvh set owner to @lvh|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_12"></a>|@lvh commented|
|-|-|

(In [40587]) Branching to 'remove-test-import-2053'

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_13"></a>|@lvh commented|
|-|-|

Oops. Sorry this took so long: the ticket is [#2053](https://github.com/twisted/twisted/issues/2053), but the patch is called 2503, which confused me, svn, trac etc terribly :)

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_14"></a>|@lvh commented|
|-|-|

Hi ewong,


1. Your patch has the wrong number. That's not so bad.
2. Your patch's topfile has the wrong number. I missed this, so I accidentally closed the wrong ticket, which made SVN/trac integration very unhappy when I tried to fix my mistake :)

Otherwise, this looks great. Buildbots are happy (modulo spurious windows failures that have nothing to do with this ticket) and I'm happy except for the part where I had to fight SVN ;)

Thanks again for your contribution

lvh

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_15"></a>|@lvh set status to `closed`|
|-|-|

(In [40593]) Merge remove-test-import-2053: remove test_import module

Author: ewong
Reviewer: lvh
Fixes: [#2053](https://github.com/twisted/twisted/issues/2053)

twisted.test.test_import was a vestigial module from back when Twisted had really bad test coverage. All it did was try to import (some of the) modules, and see if that raised any exceptions. It didn't really catch anything besides syntax errors or circular imports. Since all of the modules it imported have way better tests now, we can just go ahead and remove this.
